style/ruby/README.md

```diff
@@ -411,3 +424,239 @@ hash = { one: 1, two: 2, three: 3 }
 # também é bom, os símbolos não podem usar hífens
 hash = { "twenty" => 20, "twenty-one" => 21 }
 ```
+
+# Operadores e Fluxo de Controle
+
+## Não use `and` ou `or`
+`and` e `or` apesar de funcionar, evite usa-las. Elas não são um sinônimo dos operadores `&&` e `||`, então não devem ser usadas dessa maneira. Para expressões booleanas, use **sempre** `&&` e `||`.
+
+```ruby
+# ruim
+# e/ou em condições (sua precedência é baixa, pode produzir resultado inesperado)
+if got_needed_arguments and arguments_valid
+  # ...body omitted
+end
+
+# no cálculo da expressão lógica
+ok = got_needed_arguments and arguments_valid
+
+# bom
+# &&/|| em condicionais
+if got_needed_arguments && arguments_valid
+  # ...body omitted
+end
+# no cálculo da expressão lógica
+ok = got_needed_arguments && arguments_valid
+
+```
+
+## Não deve haver espaço depois do `!`
+
+Sem espaço após `!`.
+
+```ruby
+#ruim
+# Verifica se o array não está vazio
+if ! array.empty?
+  puts "O array tem #{array.size} elementos"
+end
+
+# bom
+# Verifica se o array não está vazio
+if !array.empty?
+  puts "O array tem #{array.size} elementos"
+end
+
+```
+
+## Dupla Negação
+
+Evite o uso de `!!`.
+
+`!!` converte um valor para booleano, mas você não precisa dessa conversão explícita na condição de uma expressão de controle; usá-la só torna sua intenção mais complexa. Se você quer fazer uma verificação de `nil`, use `nil?` em vez disso.
+
+```ruby
+# ruim
+x = 'test'
+# cheque nulo obscuro
+if !!x
+  # body omitted
+end
+
+# bom
+x = 'test'
+if x
+  # body omitted
+end
+```
+
+## Operador ternário (`?:`) vs `if`
+
+Prefira o operador ternário (`?:`) sobre as construções `if/then/else/end`. É mais comum e obviamente mais conciso.
```

Comment:
>A vantagem do operador ternário é que ele é mais conciso e pode tornar o código mais legível em algumas situações. Porém, em casos de expressões mais complexas, pode ser mais fácil e legível usar a instrução `if`.
Uma desvantagem do operador ternário é que ele pode tornar o código difícil de ler se for usado com muita frequência ou com expressões muito longas. Nesses casos, é melhor usar a instrução `if` para tornar o código mais claro e legível.
Outra diferença importante é que o operador ternário sempre retorna um valor, enquanto a instrução `if` pode ter um bloco de código que não retorna um valor. Isso significa que o operador ternário não é adequado para casos em que você precisa executar várias ações com base em uma condição, e sim apenas para casos em que você precisa retornar um valor com base em uma condição.
https://rubystyle.guide/#ternary-operator